pkg/settings/setting.go

```diff
@@ -256,3 +257,28 @@ func GetSettingByID(id string) string {
 	}
 	return provider.Get(id)
 }
+
+func MergeSettingsWithAgentEnvVars(envVars []v1.EnvVar) {
+	toCopy := []Setting{
+		ServerVersion,
+		InstallUUID,
+		IngressIPDomain,
+	}
+
+	envVarMap := map[string]v1.EnvVar{}
```

Comment:
Looks like we need this map to act as a set (we don't need values, just keys). In this case, I would recommend using `map[string]bool`. There is also an option of `map[string]struct{}` but this SO comment sums up nicely why it may be overkill: https://stackoverflow.com/a/57621291